pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -212,6 +212,17 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 		NamespacedTaskKind: true,
 		ClusterTaskRefKind: true,
 	}
+
+	if pt.OnError != "" {
+		errs = errs.Also(version.ValidateEnabledAPIFields(ctx, "OnError", config.AlphaAPIFields))
```

Comment:
are we going to use per feature flag for all new features? @JeromeJu 
from https://github.com/tektoncd/community/pull/1034

Reply:
Great question!

@JeromeJu is there any implementation that is required for per-feature flags that @QuanZhang-William can use? Do we need any implementation from the Testing end as well (may be in the plumbing repo)? 

Until those are available, we may have to continue with the existing mode of feature-flags so that we don't block new features. We could migrate this to a per-feature flag in a follow-up PR as well once all the tools are available.

WDYT?

Reply:
Yeah thanks for the question and the quick response @Yongxuanzhang and @chitrangpatel .
For now, the TEP0138 hasn't been merged yet. It would be ideal to have used the per-feature flag validation referring to [the implementation PR of TEP0138](https://github.com/tektoncd/pipeline/pull/7090). But as @chitrangpatel has put here, we might just continue to use `enable-api-fields` validations for now and I could help add those features that have been introduced between now and the policy changes after the merge of TEP0138 under the struct of per-feature flag.

Reply:
Thanks for the question and +1 to what @chitrangpatel and @JeromeJu said. I don't think this is the right time to leverage per feature flag as the proposal is not merged yet. I'm happy to migrate the feature to per feature flag model once ready. It shouldn't be a big amount of work.

Reply:
Thank you all!